### PR TITLE
Bugfix/programming exercise update/checkout repository of sample solution

### PR DESCRIPTION
**Pull request by tobiasjungmann**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context
When using the Haskell template for generating a programming exercise, the option "Checkout repository of sample solution" should be checked automatically since the Haskell testing framework relies on it. Issue: #3094

### Steps for Testing
1.	Generate a new programming exercise
2.	Select Haskell as a programming language
3.	The option "Checkout repository of sample solution" is checked.

### Screenshots
image

### CODE REVIEW

**Review by willergeorg — APPROVED** [review]
Tested on TS1. Option was checked.

**Review by NicolasUllmann — APPROVED** [review]
Tested on Server 1. Worked as expected.

**Review by canyelten — APPROVED** [review]
Tested on ts1. The option was checked as expected.

**Review by katjanakosic — APPROVED** [review]
Worked as expected :)

**Review by sinangunduz — CHANGES_REQUESTED** [review]
tested on ts3, the option was not checked.

**Review by real-dmy — APPROVED** [review]
Tested on TS1. Option was checked.

**Review by lschlesinger — CHANGES_REQUESTED** [review]
**EDIT:** I think, what I describe below is intended behavior, as one should not change the selection on `checkoutSolutionRepository` after having created the exam - It cannot be reverted anyways, I guess... 
But I would find it useful, to see the checked option in a read-only mode on the edit-page...(otherwise the information is lost, that this option was checked during exercise creation, if I am correct). But that could be out of scope for the PR :) So I guess, the PR can be approved.

Original Comment:

Tested on TS3: When generating the programming exercise for the first time with haskell as selected language, it works as expected and the option is checked. Although, after saving and reenter the edit mode (by click on edit), these selection is not present? The option does not exist and is not checked, as can be seen in the screenshot.
Bildschirmfoto 2021-04-23 um 10 47 56

**Review by lschlesinger — APPROVED** [review]
**EDIT:** I think, what I describe below is intended behavior, as one should not change the selection on `checkoutSolutionRepository` after having created the exam - It cannot be reverted anyways, I guess... 
But I would find it useful, to see the checked option in a read-only mode on the edit-page...(otherwise the information is lost, that this option was checked during exercise creation, if I am correct). But that could be out of scope for the PR :) So I guess, the PR can be approved.

Original Comment:

Tested on TS3: When generating the programming exercise for the first time with haskell as selected language, it works as expected and the option is checked. Although, after saving and reenter the edit mode (by click on edit), these selection is not present? The option does not exist and is not checked, as can be seen in the screenshot.
Bildschirmfoto 2021-04-23 um 10 47 56

**Review by cremertim — APPROVED** [review]
Tested on ts1. Works as expected

**Review by joschahenningsen — APPROVED** [review]
Worked nicely for me

**Review by ArtemTsurikov — COMMENTED** [review]
Works as expected on TS1

File: src/main/webapp/app/entities/programming-exercise.model.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export class ProgrammingExercise extends Exercise {
         this.allowOfflineIde = true; // default value
         this.programmingLanguage = ProgrammingLanguage.JAVA; // default value
         this.noVersionControlAndContinuousIntegrationAvailable = false; // default value
-        this.checkoutSolutionRepository = false; // default value
+        this.checkoutSolutionRepository = true; // default value
```
Comment (derLalla) [inline src/main/webapp/app/entities/programming-exercise.model.ts:68]: This breaks the creation of all other programming language templates.

**Pull request merged**